REBASE.openshift.md

```diff
@@ -413,25 +413,79 @@ After the initial bump as described above it is possible to update
-7. Run `make update` see [Updating generated files](#updating-generated-files).
-   NOTE: Starting from 4.10, there's a problem running `make update`, so before
-   running that `podman` command, edit `staging/src/k8s.io/code-generator/go.mod`
-   file removing this line: `k8s.io/code-generator => ../code-generator`.
+2. (optional) Revert any commits that were merged into kubernetes between previous
+   update and current one.
+3. Fetch latest state of openshift fork, checkout the appropriate branch and
+   create a new branch for the bump
+```
+git fetch openshift
+git checkout openshift/release-4.12
+git checkout -b bump-1.25.2
+```
+where `openshift` points at https://github.com/openshift/kubernetes/.
+3. Merge the changes from appropriate [released version](https://kubernetes.io/releases/patch-releases/#detailed-release-history-for-active-branches):
+```
+git merge v1.25.2
+```
+   Most likely you'll encounter conflicts, but most are around go.sum and go.mod
+   files, coming from newer versions, but at this point in time leave the conflicts
+   as they are and continue the merge.
+```
+git add --all
+git merge --continue
+
+```
+   This should create a commit titled `Merge tag 'v1.25.2' into bump-1.25.2`.
+4. Now return to the list of conflicts from previous step and fix all the files
+   picking appropriate changes, in most cases picking the newer version.
+   When done, commit all of them as another commit:
+```
+git add --all
+git commit -m "UPSTREAM: <drop>: manually resolve conflicts"
+```
+   This ensures the person reviewing the bump can easily review all the conflicts
+   and their resolution.
+5. (optional) Update openshift dependencies and run `go mod tidy` to have the
+   branch names resolved to proper go mod version. Remember to use the released
+   versions matching the branch you're modifying.
+6. Run `/bin/bash` in a container using the command and image described in [Updating generated files](#updating-generated-files)
+   section:
+```
+podman run -it --rm -v $( pwd ):/go/k8s.io/kubernetes:Z --workdir=/go/k8s.io/kubernetes registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.19-openshift-4.12 /bin/bash
+```
+   In the container run `hack/update-vendor.sh` and `make update OS_RUN_WITHOUT_DOCKER=yes`.
+7. Update kubernetes version in `openshift-hack/images/hyperkube/Dockerfile.rhel`
+   and commit all of that as:
+```
+git commit -m "UPSTREAM: <drop>: hack/update-vendor.sh, make update and update image"
+```
+8. Congratulations, you can open a PR with updated k8s patch version!
+
+### Potential problems
+
+While running `make update` in step 6 above, you might encounter one of the following problems:
 
+```
+go: inconsistent vendoring in /go/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/code-generator:
+```
+To solve it, edit `staging/src/k8s.io/code-generator/go.mod` remoing this line: `k8s.io/code-generator => ../code-generator`.
```

Comment:
AFAIU we should first try `go mod tidy` and `go mod vendor` without deleting this line. If that's right, we should change the order here.

Reply:
It's the other way around, usually removing the line is sufficient, b/c not always deps are updated. That's why I first put the modification and `go mod tidy & vendor` as a followup.

Reply:
I've also added info about bringing back that line after you're done.